### ISSUE TITLE
.next_sequence_index not found error using pre-sequence index defaults

### ISSUE THREAD

**Issue by philipstarkey** [labscript-suite-oldfinal1/runmanager#67]
**Original report ([archived issue](https://labscript-suite.github.io/labscript-suite-bitbucket-archive/#!/labscript_suite/runmanager/issues/67)) by David Meyer (Bitbucket: [dihm](https://bitbucket.org/%7B8df655d6-8661-4957-b4ba-669cef19bf2d%7D/), GitHub: [dihm](https://github.com/dihm)).**

----------------------------------------
Updating to tag 2.4.0 I now get many file not found errors for the .next_sequence_index file when opening and closing runmanager. It also appears that the closing error is preventing runmanager from updating runmanager.ini.

Also, my shot output folder gets erased on load. I get the corresponding non-default output folder message, but the reset to default folder button is not available.

I'm guessing that part of this is just me not understanding how to work with the new sequence_index, but I also thought the idea was to not force people to immediately update to the new scheme so I'm not quite sure what is going on.

The opening runmanager traceback (which appears twice) is
```
Traceback (most recent call last):
  File "C:\labscript_suite\runmanager\__main__.py", line 1411, in load_the_config_file
    self.load_configuration(autoload_config_file)
  File "C:\labscript_suite\runmanager\__main__.py", line 3110, in load_configuration
    default_output_folder = self.get_default_output_folder()
  File "C:\Anaconda3\envs\labscript\lib\site-packages\qtutils\invoke_in_main.py", line 228, in f
    return inmain(fn, *args, **kwargs)
  File "C:\Anaconda3\envs\labscript\lib\site-packages\qtutils\invoke_in_main.py", line 102, in inmain
    return fn(*args, **kwargs)
  File "C:\labscript_suite\runmanager\__main__.py", line 2388, in get_default_output_folder
    increment_sequence_index=False,
  File "C:\labscript_suite\runmanager\__init__.py", line 663, in new_sequence_details
    sequence_index = next_sequence_index(shot_basedir, now, increment_sequence_index)
  File "C:\labscript_suite\runmanager\__init__.py", line 624, in next_sequence_index
    with open(sequence_index_file) as f:
IOError: [Errno 2] No such file or directory: u'C:\\Users\\Experiment Control\\Experiments\\RydEIT2\\ProbeSweep\\.next_sequence_index'
```

Relevant labconfig is
```
[DEFAULT]
experiment_shot_storage = %(shared_drive)s\Experiments\%(experiment_name)s
[runmanager]
autoload_config_file = %(experiment_shot_storage)s\runmanager.ini
output_folder_format = %%Y\%%m\%%d
```

**Comment by philipstarkey** [comment]
**Original comment by Chris Billington (Bitbucket: [cbillington](https://bitbucket.org/%7Be363c5a9-5075-4656-afb5-88bd6a6dceeb%7D/), GitHub: [chrisjbillington](https://github.com/chrisjbillington)).**

----------------------------------------
Thanks for the bug report! 
It was supposed to be completely backward compatible, so apologies that it broke for you. You shouldn't have to understand or do anything for it not to break. I'll see what I can do.

**Comment by philipstarkey** [comment]
**Original comment by Chris Billington (Bitbucket: [cbillington](https://bitbucket.org/%7Be363c5a9-5075-4656-afb5-88bd6a6dceeb%7D/), GitHub: [chrisjbillington](https://github.com/chrisjbillington)).**

----------------------------------------
What Python version are you on? I think the issue is that different Python versions raise different errors when a file or directory does not exist, and I'm not catching the full range of possibilities when determining if the file exists or not.

**Comment by philipstarkey** [comment]
**Original comment by David Meyer (Bitbucket: [dihm](https://bitbucket.org/%7B8df655d6-8661-4957-b4ba-669cef19bf2d%7D/), GitHub: [dihm](https://github.com/dihm)).**

----------------------------------------
Oh, that's interesting. At tag 2.4.0 the first error on open gets caught and displayed in the output box, with the second in a pop-up. Going to [PR #35](https://bitbucket-archive.labscriptsuite.org/#!/labscript_suite/runmanager/pull-requests/35) both errors show as pop-ups. Not really sure what that means.

**Comment by philipstarkey** [comment]
**Original comment by David Meyer (Bitbucket: [dihm](https://bitbucket.org/%7B8df655d6-8661-4957-b4ba-669cef19bf2d%7D/), GitHub: [dihm](https://github.com/dihm)).**

----------------------------------------
I'm python 2.7.15 on this system.

**Comment by philipstarkey** [comment]
**Original comment by Chris Billington (Bitbucket: [cbillington](https://bitbucket.org/%7Be363c5a9-5075-4656-afb5-88bd6a6dceeb%7D/), GitHub: [chrisjbillington](https://github.com/chrisjbillington)).**

----------------------------------------
Oh, yikes, the change in how the errors appear is my fault - I left debug code in the PR that raised the error instead of showing it in the output box. Basically if errors occur during config-file-loading they are displayed in the outputbox since the are probably about the config file. But during developing it hides actual errors so I turned that off and forgot to turn it back on again...

Great, it's definitely that I'm not catching the right exceptions. I'll read up on the exception hierarchy changes for OSErrors etc between Python 2 and 3 and make sure I'm catching the right exception. It will be an easy fix, just gotta make sure I'm covering all the bases. Will have it fixed soon!

**Comment by philipstarkey** [comment]
**Original comment by Chris Billington (Bitbucket: [cbillington](https://bitbucket.org/%7Be363c5a9-5075-4656-afb5-88bd6a6dceeb%7D/), GitHub: [chrisjbillington](https://github.com/chrisjbillington)).**

----------------------------------------
This should be resolved by b2ca5d9ed415c65a82261761a447989efa586a61, could you check if it fixes it as expected?

**Comment by philipstarkey** [comment]
**Original comment by David Meyer (Bitbucket: [dihm](https://bitbucket.org/%7B8df655d6-8661-4957-b4ba-669cef19bf2d%7D/), GitHub: [dihm](https://github.com/dihm)).**

----------------------------------------
3e0aa63 works like a charm.

**Comment by philipstarkey** [comment]
**Original comment by Chris Billington (Bitbucket: [cbillington](https://bitbucket.org/%7Be363c5a9-5075-4656-afb5-88bd6a6dceeb%7D/), GitHub: [chrisjbillington](https://github.com/chrisjbillington)).**

----------------------------------------

* changed state from "new" to "resolved"

Fixed in b2ca5d9ed415c65a82261761a447989efa586a61
